### PR TITLE
Modify the packages and patterns to support arch conditional

### PR DESCRIPTION
**Pull request by teclator**
## Problem

The current configuration does not have arch conditional for the patterns or packages needed by each product, as well as we do not have a general support for it.

## Solution

Extend the current software installation repositories approach supporting the arch conditional also in the software selection

## Testing

- *Added a new unit test*
- *Tested manually*

### CODE REVIEW

File: service/lib/dinstaller/software/manager.rb [hunk 1]
```diff
@@ -212,25 +212,27 @@ def import_gpg_keys
...
+      def arch_collection_for(section, key)
+        arch_select(section).map { |r| r.is_a?(Hash) ? r[key] : r }
       end
 
       def selected_base_product
         @config.data["software"]["base_product"]
       end
 
+      def arch_match?(archs)
+        return true if archs.nil?
```
Comment (imobachgs) [inline service/lib/dinstaller/software/manager.rb:229]: `nil?` or `empty?`?

Comment (imobachgs) [reply service/lib/dinstaller/software/manager.rb:229]: Just forget about it. It is fine as it is.

**Review by imobachgs — APPROVED** [review]

File: service/package/rubygem-d-installer.changes [hunk 2]
```diff
@@ -1,3 +1,9 @@
+-------------------------------------------------------------------
+Tue Mar 21 21:51:48 UTC 2023 - Knut Anderssen <kanderssen@suse.com>
+
+- Allow to define architecture specific patterns and packages
+  (gh#yast/d-installer#86).
```
Comment (dgdavid) [inline service/package/rubygem-d-installer.changes:5]: #86 or #486 ?

**Review by dgdavid — CHANGES_REQUESTED** [review]

File: service/package/rubygem-d-installer.changes (hunk 2 above)
Comment (imobachgs) [reply service/package/rubygem-d-installer.changes:5]: Oh, you are right, the number is wrong.

**Review by imobachgs — APPROVED** [review]

**Pull request merged**